sdk/storage/inc/datalake/datalake_options.hpp

```diff
@@ -0,0 +1,943 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "common/shared_request_options.hpp"
+#include "protocol/datalake_rest_client.hpp"
+
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Storage { namespace DataLake {
+
+  /**
+   * @brief Service client options used to initalize DataLakeServiceClient.
+   */
+  struct ServiceClientOptions
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+  };
+
+  /**
+   * @brief File system client options used to initalize DataLakeFileSystemClient.
+   */
+  struct FileSystemClientOptions
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+  };
+
+  /**
+   * @brief Path client options used to initalize DataLakePathClient.
+   */
+  struct PathClientOptions
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+  };
+
+  /**
+   * @brief Optional parameters for DataLakeServiceClient::ListFilesSystems
+   */
+  struct ListFileSystemsOptions : public SharedRequestOptions
+  {
+    /**
+     * @brief Filters results to filesystems within the specified prefix.
+     */
+    std::string Prefix;
+
+    /**
+     * @brief The number of filesystems returned with each invocation is
+     *        limited. If the number of filesystems to be returned exceeds
+     *        this limit, a continuation token is returned in the response
+     *        header x-ms-continuation. When a continuation token is returned
+     *        in the response, it must be specified in a subsequent invocation
+     *        of the list operation to continue listing the filesystems.
+     */
+    std::string Continuation;
+
+    /**
+     * @brief An optional value that specifies the maximum number of items to
+     *        return. If omitted or greater than 5,000, the response will
+     *        include up to 5,000 items.
+     */
+    uint32_t MaxResults = 0;
```

Comment:
did we agreed on using only signed variables?

Reply:
Nice catch, I'll change it.